kong/plugins/api-response-merger/body_transformer.lua

```diff
@@ -139,6 +139,9 @@ local function set_for_path(data_path, upstream_body, value, resource)
 
   --- if we are using json.path to set result
   local json_path_set = false
+  if resource ~= nil and resource.config.add_missing == false then
+    json_path_set = true
```

Comment:
IMO we should add utils.is_array(upstream_body, 'fast')
```
      if utils.is_array(current, 'fast')  and current[path[i]] ~= nil then
```

https://github.com/inveox-lab-it/kong-plugin-api-response-merger/blob/38f1f4144f821919396e98e7f59d9052009d59b3/kong/plugins/api-response-merger/body_transformer.lua#L198

Reply:
This is not a matter of whether current is an array - rather a table. And the problem lies in calling this part for paths like this $..statusChanges..modifiedBy.
And this is only additional check of the symptom, not the reason behind the problem (which is calling this path for multiple data_paths in which some of them were resolved and others not)

Reply:
In previously developing cycles error in this place was caused by this call https://github.com/inveox-lab-it/kong-plugin-api-response-merger/blob/master/kong/plugins/api-response-merger/handler.lua#L107

I'm not sure how setting `json_path_set` flag to true will help (except that we can skip setting of data)